plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/AbstractResourceMojo.java

```diff
@@ -92,30 +93,54 @@ protected void writeResources(KubernetesList resources, ResourceClassifier class
         }
     }
 
-    public static File writeResourcesIndividualAndComposite(KubernetesList resources, File resourceFileBase, ResourceFileType resourceFileType, Logger log) throws MojoExecutionException {
+    public static File writeResourcesIndividualAndComposite(KubernetesList resources, File resourceFileBase, ResourceFileType resourceFileType, Logger log, Boolean generateRoute) throws MojoExecutionException {
+
+        List<HasMetadata> oldItemList = resources.getItems();
```

Comment:
I think possibly we can get rid of this `oldItemList`; Since you're simply modifying resource object in your route generate condition. Or is there something I'm missing here?

Reply:
Here I am storing the old list because I am deleting the Route for generating openshift.yml. Later I will need all the resources and that's why it is again added to resources [here](https://github.com/fabric8io/fabric8-maven-plugin/pull/1211/files/4774dd63e61ab3c70b64e6e898112ff53346aa12#diff-cc525cbb3a0475474fadc5ce80911c1cR126)